src/coins.cpp

```diff
@@ -1330,6 +1327,30 @@ bool CCoinsViewCache::CheckScTxTiming(const uint256& scId) const
     return true;
 }
 
+/**
+ * @brief Checks whether a Forward Transfer output is still valid based on sidechain current FT fee.
+ * 
+ * @param ftOutput The Forward Transfer output to be checked.
+ * @return true if ftOutput is still valid, false otherwise.
+ */
+bool CCoinsViewCache::CheckScFtFee(const CTxForwardTransferOut& ftOutput) const
+{
+    CScCertificateView certView = GetActiveCertView(ftOutput.scId);
+    return ftOutput.nValue > certView.forwardTransferScFee;
+}
+
+/**
+ * @brief Checks whether a Mainchain Backward Transfer Request output is still valid based on sidechain current MBTR fee.
+ * 
+ * @param mbtrOutput The Mainchain Backward Transfer Request output to be checked.
+ * @return true if mbtrOutput is still valid, false otherwise.
+ */
+bool CCoinsViewCache::CheckScMbtrFee(const CBwtRequestOut& mbtrOutput) const
+{
+    CScCertificateView certView = GetActiveCertView(mbtrOutput.scId);
+    return mbtrOutput.scFee >= certView.mainchainBackwardTransferRequestScFee;
+}
+
 CValidationState::Code CCoinsViewCache::IsScTxApplicableToState(const CTransaction& tx) const
```

Comment:
introduce an optional bool ptr to indicate to caller if the type of error should cause to ban the sender (this will be typically used by accepttomempool because for blocks, every error should cause to ban the sender

Reply:
Done